packages/bcer-api/app/src/sales/sales.controller.ts

```diff
@@ -62,6 +78,112 @@ export class SalesReportController {
     @Request() req: RequestWithUser,
   ): Promise<SalesReportRO[]> {
     const sales = await this.service.getSalesReports();
-    return sales.map((salesReport: SalesReportEntity) => salesReport.toResponseObject());
+    return sales.map((salesReport: SalesReportEntity) =>
+      salesReport.toResponseObject(),
+    );
+  }
+
+  @ApiOperation({ summary: 'Create Sales Reports from Submission' })
+  @ApiResponse({ status: HttpStatus.CREATED, type: [SalesReportRO] })
+  @HttpCode(HttpStatus.CREATED)
+  @Roles('user')
+  @UseGuards(BusinessGuard)
+  @Post('/:submissionId')
+  async createProductSoldSalesReports(
+    @Param('submissionId') submissionId: string,
+    @Request() req: RequestWithUser,
+  ): Promise<any> {
+    const submission = await this.submissionService.getOne(submissionId);
+
+    if (submission.businessId !== req.ctx.businessId) {
+      throw new ForbiddenException(
+        'This user does not have access to this submission',
+      );
+    }
+
+    const saleReportsData = submission.data.saleReports;
+    const locationId = submission.data.locationId;
+    const year = submission.data.year;
+    const businessId = submission.businessId;
+
+    // validation
```

Comment:
Could you elaborate on this with a comment, what are the requirements for these?

Reply:
This is for the validation of the csv, it has some restrictions. 
![image](https://user-images.githubusercontent.com/46575401/134402917-0436d70b-78fe-4466-8134-0bf31b8ff501.png)